app/models/order.rb

```diff
@@ -1,3 +1,27 @@
-class Order < ApplicationRecord
+class Order < ApplicationRecord  
   belongs_to :user
+  has_many :orders_items
+  has_many :items, through: :orders_items
+
+  enum status: %w(ordered paid cancelled completed)
+
+  def total_price
+    orders_items.map { |item| item.current_price_per_unit * item.quantity.to_f }.sum
+  end
+
+  def item_quantity(item)
+    orders_items.where(item_id: item.id).count
+  end
+
+  def subtotal(item)
```

Comment:
@wlffann Can we move this logic into `order_item`. `#subtotal`can be calculated on the `order_item` level.